### PR TITLE
chore(interpreter): Support object replies

### LINKED ISSUE #2880: LUA script issue in Dragonfly for HGETALL result
Hi all,

**Describe the bug**
I am trying to go from Redis to Dragonfly. I am using a set of scripts to automate my work in the database.
One of my script is behaving differently in Dragonfly than in Redis and does not provide the expected result.

**To Reproduce**
1. Insert records using `HSET id:99 value 99`
2. Load script using `SCRIPT LOAD "#!lua flags=allow-undeclared-keys\nlocal result = {map={}}\nresult.map[\"id:99\"] = redis.call(\"HGETALL\", \"id:99\")\nreturn {1, result}"`
3. Call script using `EVALSHA <previsously returned SHA> 0 0`
4. See empty result:
```
   1) (integer) 1
   2) (empty array)
```

When running in Redis `#!lua flags=allow-undeclared-keys\n` should be removed from the script.

**Expected behavior**
This should return the same as Redis:
  ```
1) (integer) 1
  2) 1) "id:99"
      2) 1) "value"
          2) "99"
```

**Screenshots**
In Redis:
image
In Dragonfly:
image

**Environment (please complete the following information):**
 - Kubernetes using dragonfly-operator v1.1.2 using helm chart from https://github.com/dragonflydb/dragonfly-operator/pull/174
 - Dragonfly Version: [e.g. 1.6.0]

### PR DESCRIPTION
**Pull request by dranikpg**
Closes #2880

### CODE REVIEW

File: src/core/interpreter.cc [hunk 1]
```diff
@@ -757,40 +758,34 @@ bool Interpreter::IsTableSafe() const {
...
-
-        // reset to iterate on the next table.
-        i = 0;
-        len = lua_rawlen(lua_, -1);
-      } else {
-        lua_pop(lua_, 1);  // pop table element
-        ++i;
-      }
-    }
+  // Copy root table because we remove it upon finishing traveral
```
Comment (romange) [inline src/core/interpreter.cc:761]: traveral -> traversal

File: src/core/interpreter.cc [hunk 2]
```diff
@@ -757,40 +758,34 @@ bool Interpreter::IsTableSafe() const {
...
-    lua_pop(lua_, 1);
-  };
+    bool interrupted = false;
+    for (; lua_next(lua_, -2) != 0; lua_pop(lua_, 1)) {
+      if (lua_type(lua_, -1) != LUA_TTABLE)
+        continue;
+
+      depth++;
+      lua_pushnil(lua_);
+      interrupted = true;
```
Comment (romange) [inline src/core/interpreter.cc:780]: can you please add a comment, what does it mean interrupted?

Comment (dranikpg) [reply src/core/interpreter.cc]: also changed it to descending, bad naming

**Review by romange — APPROVED** [review]
👏🏼

**Comment by aloubiex38** [comment]
Thanks guys.

**Pull request merged**